### PR TITLE
Ft login

### LINKED ISSUE #2: [APP] Create login user screen
- [x] Create login layout
- [ ] Integrate with backend ([documentation](https://set-api-production.up.railway.app/api#/default/AuthController_makeLogin))
- [x] Store the user info in Context API (or other client state management tool)
- [ ] (Optional) - Store the user info in [Async Storage](https://docs.expo.dev/versions/latest/sdk/async-storage/)

Login

### PR DESCRIPTION
**Pull request by RodrigoRVSN**
Closes #2

### CODE REVIEW

**Review by gianlucabueno — APPROVED** [review]

File: src/views/Welcome/index.tsx [hunk 1]
```diff
@@ -0,0 +1,44 @@
+/* eslint-disable @typescript-eslint/no-unsafe-assignment */
+import { useNavigation } from '@react-navigation/native'
+import React from 'react'
+import { View, Text, Image, TouchableOpacity } from 'react-native'
+import * as Animatable from 'react-native-animatable'
+import { styles } from './styles'
+
+export default function Welcome() {
```
Comment (RodrigoRVSN) [inline src/views/Welcome/index.tsx:8]: **question:** ‏this screen is new?

File: src/views/SignIn/styles.ts [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+  },
+  input: {
+    borderBottomWidth: 1,
+    height: 40,
+    marginBottom: 12,
+    fontSize: 16
+  },
+  button: {
+    backgroundColor: '#38A69D',
+    width: '40%',
```
Comment (RodrigoRVSN) [inline src/views/SignIn/styles.ts:38]: tip: avoid magic numbers

File: src/views/SignIn/index.tsx [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+  password: string
+}
+
+export default function SignIn() {
+  const { control, handleSubmit, formState: { errors } } = useForm<FormData>()
+  const { signed, signIn } = useAuth()
+
+  const handleSignIn = (data: FormData) => {
+    const { email, password } = data
+    if (email && password) signIn(email, password)
```
Comment (RodrigoRVSN) [inline src/views/SignIn/index.tsx:19]:
```suggestion
  const handleSignIn = ({ email, password }: FormData) => {
    if (email && password) signIn(email, password)
```

File: src/views/SignIn/index.tsx [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+              placeholder='Digite sua senha'
+              autoCapitalize='none'
+              secureTextEntry
+              style={styles.input}
+              value={value}
+              onChangeText={onChange}
+            />
+          )}
+        />
+        {errors?.password && <Text style={styles.errorText}>Senha obrigatória!</Text>}
```
Comment (RodrigoRVSN) [inline src/views/SignIn/index.tsx:61]: **nitpick:** ‏having error !== password missing

File: src/services/Auth/auth.ts [hunk 5]
```diff
@@ -0,0 +1,13 @@
+import api from "../Api/api";
+
+export async function signIn (email: string, password: string) {
+    try {
+      const response = await api.post('/user/login', {
+        email: email,
+        password: password,
```
Comment (RodrigoRVSN) [inline src/services/Auth/auth.ts:7]: object key = value

```suggestion
        email,
        password,
```

File: src/routes/routes.tsx [hunk 6]
```diff
@@ -0,0 +1,21 @@
+import React, { useContext } from 'react'
+import { View, ActivityIndicator } from 'react-native'
+import {useAuth} from '../contexts/auth'
+import AppRoutes from './app.routes'
+import AuthRoutes from './auth.routes'
+
+
+const Routes: React.FC = () => {
```
Comment (RodrigoRVSN) [inline src/routes/routes.tsx:8]: nitpick: we could use typescript inference (same about others components)

```suggestion
const Routes = () => {
```

File: src/contexts/auth.tsx [hunk 7]
```diff
@@ -0,0 +1,54 @@
...
+
+  //   }
+  //   loadStorageData()
+  // }, [])
+
+  async function signIn(email: string, password: string): Promise<void> {
+    try {
+      const response = await auth.signIn(email, password)
+      setUser(response.user)
+      // await AsyncStorage.setItem('@SETAuth:user', JSON.stringify(response.user))
```
Comment (RodrigoRVSN) [inline src/contexts/auth.tsx:35]: **suggestion:** ‏we could open a new issue in this repo to add the `AsyncStorage` functionality and remove this comments too

File: src/contexts/auth.tsx [hunk 8]
```diff
@@ -0,0 +1,54 @@
...
+  loading: boolean,
+  signIn(email: string, password: string): Promise<void>,
+  signOut(): void
+}
+
+const AuthContext = createContext<AuthContextData>({} as AuthContextData)
+
+export const AuthProvider = ({ children }) => {
+  const [user, setUser] = useState<object | null>(null)
+  const [loading, setLoading] = useState<boolean>(false)
```
Comment (RodrigoRVSN) [inline src/contexts/auth.tsx:17]: suggestion: we could use inference of typescript

```suggestion
  const [loading, setLoading] = useState(false)
```

**Review by RodrigoRVSN — COMMENTED** [review]
Nenhum dos comentarios abaixo é bloqueante, consegue mergear @mayoral-leonardo? Ou eu que preciso mergear? Se tu n tiver permissao me avisa que mergeio a PR

**Comment by RodrigoRVSN** [comment]
Por mim tá ok, só mergear / da um salve para eu mergear 👍🏻

**Review by mayoral-leonardo — COMMENTED** [review]
PR merge.

**Pull request merged**